### PR TITLE
server: Add an inspect endpoint for containers

### PR DESCRIPTION
**Pull request by mrunalp**
We can add IP once #814 is merged. 
This could be used by our cadvisor handler.

Signed-off-by: Mrunal Patel <mpatel@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
lint failures

File: server/inspect.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			http.Error(w, fmt.Sprintf("container %s state is nil", containerId), http.StatusNotFound)
+			return
+		}
+
+		ci := ContainerInfo{
+			Pid:         ctrState.Pid,
+			Image:       ctr.Image(),
+			CreatedTime: ctrState.Created.UnixNano(),
+			Labels:      ctr.Labels(),
+		}
```
Comment (runcom) [inline server/inspect.go]: we do need annotations https://github.com/derekwaynecarr/kubernetes/pull/4/files#diff-84f52d75d93d7dd10d29024469937e2eR211 or get `restartCount` from container itself

File: server/inspect.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "applicaton/json")
+		w.Write(js)
+	}))
+
+	// TODO: Make this configurable
+	return http.ListenAndServe(":7373", mux)
```
Comment (runcom) [inline server/inspect.go]: is this defaulting to listen just on localhost?

**Comment by runcom** [comment]
I'll switch file readings from https://github.com/derekwaynecarr/kubernetes/pull/4 to calling this new route

File: server/inspect.go (hunk 1 above)
Comment (mrunalp) [reply server/inspect.go]: Added annotations.

**Comment by mrunalp** [comment]
Pushed update.

File: server/inspect.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	"net/http"
+
+	"github.com/go-zoo/bone"
+)
+
+// ContainerInfo stores information about containers
+type ContainerInfo struct {
+	Pid         int               `json:"pid"`
+	Image       string            `json:"image"`
+	CreatedTime int64             `json:"created_time"`
```
Comment (runcom) [inline server/inspect.go]: can we format it to `time.RFC3339Nano`? cadvisor is gonna read into that and is more human readable for people calling the route

Comment (mrunalp) [reply server/inspect.go]: We could convert to RFC3339Nano in cadvisor. (Just to get this in for now and then can modify it.)

**Comment by mrunalp** [comment]
retest

File: server/inspect.go (hunk 2 above)
Comment (mrunalp) [reply server/inspect.go]: localhost:7373/containers/id should work for this.

Comment (runcom) [reply server/inspect.go]: Yeah, let's just make sure we don't expose anything (and afaict it just binds on localhost)

**Comment by runcom** [comment]
LGTM after CI

File: server/inspect.go (hunk 2 above)
Comment (mrunalp) [reply server/inspect.go]: @rhatdan I agree it should be restricted to localhost. Let me double check and ensure that is the case. Thanks!

**Comment by mrunalp** [comment]
retest

File: server/inspect.go (hunk 2 above)
Comment (mrunalp) [reply server/inspect.go]: okay, it was listening on all interface so I pushed an update to explicitly specify localhost:7373.

**Comment by runcom** [comment]
what's wrong here -.-

**Comment by mrunalp** [comment]
lint out of memory :/ restarting job.

**Comment by mrunalp** [comment]
[retest]

**Comment by mrunalp** [comment]
Tests passed. Merging.

**Pull request merged**